spec/requests/api/providers_spec.rb

```diff
@@ -130,6 +146,24 @@
       expect(response).to have_http_status(:forbidden)
     end
 
+    it "creates provider with custom attributes" do
+      api_basic_authorize collection_action_identifier(:providers, :create)
+      run_post(providers_url, sample_rhevm.merge!("custom_attributes" => custom_attributes))
+      provider = ExtManagementSystem.find_by(:name => "sample rhevm")
+
+      expect(provider.custom_attributes.count).to eq(3)
+
+      expect(CustomAttribute.find_by_name("experation_date").serialized_value.class).to eq(Date)
```

Comment:
As elsewhere, this is testing low level behavior in a request spec - it would be better tested against the response body or, if it's not exposed, ensure that the model has sufficient coverage for this behavior